lib/string.gi

```diff
@@ -1258,9 +1258,121 @@ InstallGlobalFunction(StringOfMemoryAmount, function(m)
+        type := "s";
+      fi;
+      return rec(id := string{[startpos..posbang-1]}, type := type);
+    end;
+      
+    argcounter := 1;
+    len := Length(s);
+    pos := 0;
+
+    if not (IsOutputStream(stream) or IsString(stream)) or not IsString(s) then
+        ErrorNoReturn("Usage: PrintToFormatted(<stream>, <string>, <args>...)");
+    fi;
+
+    while pos < len do
+        nextbrace := Position(s, '{', pos);
+        endbrace := Position(s, '}', pos);
+        # Keep checking for '}}' until we find an '{'
+        while endbrace <> fail and endbrace < nextbrace do
+            if endbrace + 1 <= len and s[endbrace + 1] = '}' then
+                AppendTo(stream, s{[pos+1..endbrace]});
+                pos := endbrace + 1;
+                endbrace := Position(s, '}', pos);
+            else
+                ErrorNoReturn("Mismatched '}' at position ",endbrace);
+            fi;
+        od;
+
+        if nextbrace = fail then
+            AppendTo(stream, s{[pos+1..len]});
+            return;
+        fi;
+
+        AppendTo(stream, s{[pos+1..nextbrace-1]});
+
+        # If this is {{, then print a { and call 'continue'
+        if nextbrace+1 <= len and s[nextbrace+1] = '{' then
+            AppendTo(stream, "{");
+            pos := nextbrace + 1;
+            continue;
+        fi;
+
+        if endbrace = fail then
+            ErrorNoReturn("Invalid format string, no matching '}' at position ", nextbrace);
+        fi;
+
+        toprint := splitBracket(s, nextbrace+1,endbrace-1);
+
+
+        if toprint.id = "" then
+            if Length(args) < argcounter then
+                ErrorNoReturn("out of bounds in StringFormatted -- used ",argcounter," {} when there are only ",Length(args), " arguments");
+            fi;
+            var := args[argcounter];
+            argcounter := argcounter + 1;
+        elif Int(toprint.id) <> fail then
+            if Int(toprint.id) < 1 or Int(toprint.id) > Length(args) then
+                ErrorNoReturn("out of bounds in StringFormatted -- asked for {",Int(toprint.id),"} when there are only ",Length(args), " arguments");
+            fi;
+            var := args[Int(toprint.id)];
+        else
+            if not IsRecord(args[1]) then
+                ErrorNoReturn("first data argument must be a record when using {",toprint.id,"}");
+            fi;
+            if not IsBound(args[1].(toprint.id)) then
+                ErrorNoReturn("No record member '",toprint[1].id,"'");
+            fi;
+            var := args[1].(toprint.id);
+        fi;
+        pos := endbrace;
+
+        if toprint.type = "s" then
+          if not IsString(var) then
+            var := String(var);
+          fi;
+          AppendTo(stream, var);
+        elif toprint.type = "v" then
+          AppendTo(stream, ViewString(var));
+        elif toprint.type = "d" then
+          AppendTo(stream, DisplayString(var));
+        else ErrorNoReturn("Invalid formatting option: '", toprint.type, "'");
```

Comment:
And a third name: after "format" and "type" this is now called "formatting option".